### DISCUSSION TITLE
[Custom Cell] Implementing Custom Render Logic in Text Cell

### DISCUSSION THREAD

**Discussion by xhw994** [glideapps/glide-data-grid#807] [Q&A]
Hi there,

I want to have custom render logic to my text cell. Normally the cell should render just like a text cell, however, when there's an error in data, I want to render a tooltip in front of the text. The tooltip is just a ReactElement made by myself. How can I achieve this?

I'm working on a project where I need to enhance a text cell with custom rendering logic. The standard behavior of the cell is straightforward, similar to a typical text cell. However, I want to introduce a specific feature: when the cell's data contains an error, a tooltip should appear in front of the text.

This tooltip is a custom ReactElement that I've created. I'm looking for guidance on integrating this conditional rendering within the text cell. Here are my specific questions:

- What's the best practice for dynamically rendering a component (like my custom tooltip) based on this condition?
- Are there any performance considerations I should be aware of when implementing this conditional rendering?

Any code snippets, library recommendations, or general advice on this would be greatly appreciated. If more details are needed, I'm happy to provide them.

Thank you in advance for your help!

**Comment by xhw994** [comment]
For simpler use case, how can I draw any SVG within a read-only cell?

**Reply by jassmith** [reply]
You need to load it into an `image` using the ImageWindowLoader.
